typing/out_type.ml

```diff
@@ -708,13 +708,16 @@ let printer_iter_type_expr f ty =
 let quoted_ident ppf x =
   Style.as_inline_code !Oprint.out_ident ppf x
 
-module Internal_names : sig
+module Internal_names' : sig
 
   val reset : unit -> unit
 
   val add : Path.t -> unit
 
-  val print_explanations : Env.t -> Fmt.formatter -> unit
+  val print_explanations :
+    add_type_to_preparation: (type_expr -> unit) ->
+    tree_of_typexp: (type_or_scheme -> type_expr -> out_type) ->
+    Env.t -> Fmt.formatter -> unit
```

Comment:
I am wondering if it would not be clearer to expose a `eqns` and `existential_constructor` functions which would return the set of equations and constructors and then delegate the printing to `Errortrace_report`.

Reply:
I gave it a go